tests/RobotFramework/tests/cumulocity/registration/device_registration.robot

```diff
@@ -29,13 +29,117 @@ Child device registration
+
+Register nested child device using default topic schema via MQTT
+    ${child_level1}=    Get Random Name
+    ${child_level2}=    Get Random Name
+    ${child_level3}=    Get Random Name
+
+    Execute Command    tedge mqtt pub --retain 'te/device/${child_level1}//' '{"@type":"child-device","@parent":"device/main//"}'
+    Execute Command    tedge mqtt pub --retain 'te/device/${child_level2}//' '{"@type":"child-device","@parent":"device/${child_level1}//","name":"${child_level2}"}'
+    Execute Command    tedge mqtt pub --retain 'te/device/${child_level3}//' '{"@type":"child-device","@parent":"device/${child_level2}//","type":"child_level3"}'
+    Execute Command    tedge mqtt pub --retain 'te/device/${child_level3}/service/custom-app' '{"@type":"service","@parent":"device/${child_level3}//","name":"custom-app","type":"service-level3"}'
+
+    # Level 1
+    Check Child Device    parent_sn=${DEVICE_SN}    child_sn=${DEVICE_SN}:device:${child_level1}    child_name=${DEVICE_SN}:device:${child_level1}    child_type=thin-edge.io-child
+
+    # Level 2
+    Check Child Device    parent_sn=${DEVICE_SN}:device:${child_level1}    child_sn=${DEVICE_SN}:device:${child_level2}    child_name=${child_level2}    child_type=thin-edge.io-child
+
+    # Level 3
+    Check Child Device    parent_sn=${DEVICE_SN}:device:${child_level2}    child_sn=${DEVICE_SN}:device:${child_level3}    child_name=${DEVICE_SN}:device:${child_level3}    child_type=child_level3
+    Check Service    child_sn=${DEVICE_SN}:device:${child_level3}    service_sn=${DEVICE_SN}:device:${child_level3}:service:custom-app    service_name=custom-app    service_type=service-level3    service_status=up
+
+
+Register service on a child device via MQTT
+    Execute Command    tedge mqtt pub --retain 'te/device/${CHILD_SN}//' '{"@type":"child-device","name":"${CHILD_SN}","type":"linux-device-Aböut"}'
+    Execute Command    tedge mqtt pub --retain 'te/device/${CHILD_SN}/service/custom-app' '{"@type":"service","@parent":"device/${CHILD_SN}//","name":"custom-app","type":"custom-type"}'
+
+    # Check child registration
+    Check Child Device    parent_sn=${DEVICE_SN}    child_sn=${DEVICE_SN}:device:${CHILD_SN}    child_name=${CHILD_SN}    child_type=linux-device-Aböut
+
+    # Check service registration
+    Check Service    child_sn=${DEVICE_SN}:device:${CHILD_SN}    service_sn=${DEVICE_SN}:device:${CHILD_SN}:service:custom-app    service_name=custom-app    service_type=custom-type    service_status=up
+
+
+Register devices using custom MQTT schema
+    [Documentation]    Complex example showing how to use custom MQTT topics to register devices/services using
+        ...            custom identity schemas
+
+    # Main device registration via MQTT is not supported at the moment
+    # Execute Command    tedge mqtt pub --retain 'te/base///' '{"@type":"main-device","name":"base","type":"te_gateway"}'
```

Comment:
Even though the registration of a new main device can't be supported atm, I'm assuming that the expected behaviour here was to just add this custom topic id also into the entity store, associated to the main device's external id itself so that data to the main device can be sent via this custom topic as well. That much can be done. The `@id`, `name` and `type` fields will be ignored until we support device fragment updates as well.

Reply:
> Even though the registration of a new main device can't be supported atm

I think the issue is to create the EntityStore from a main-device registration message: this message can only be a guess of the topic id used by the main device.

The EntityStore should be created only after a main-device CN. It's registration can then be done either explicitly with a registration message or automatically if some MEA is published on `device/main///#`.

Reply:
Enabled overriding of the default schema using another registration message with custom topic scheme.